src/seedu/addressbook/Main.java

```diff
@@ -104,6 +104,10 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        if (!storage.fileExists()){
+            ui.showToUser(storage.fileDeletedWarning());
```

Comment:
You shouldn't checked the file here, this doesn't satisfy SLAP. Better to check inside `save` method. Also consider creating the file and show the problem to the user, instead of closing the program.